site/guides/tenant-onboarding.md

```diff
@@ -0,0 +1,274 @@
+isolate-network-policy.yaml  quota.large.yaml  quota.medium.yaml
+quota.small.yaml  role-binding.yaml  service-account.yaml
+
+pkg-catalog $ cat tenant/.krmignore
+.snippets
+
+```
+
+### Publishing Tenant Package
+
+So once you are happy with the tenant package, you can publish the tenant
+package by tagging the version as shown below:
+
+```shell
+# Assuming you are in the pkg catalog repo where tenant package exists
+
+# Pl. remove the inventory information from the Kptfile before publishing
+
+# create new tag
+$ git tag tenant/v0.1 main
+
+# push the tag to the upstream
+$ git push origin tenant/v0.1
+
+```
+
+## Tenant onboarding workflow
+
+Now, let’s take a look at how tenant onboarding will work. Steps described below
+can be done by a member of the platform team or application developer team. The
+good thing is that it enables self-service workflow for application teams where
+they can request for new tenants by simply issuing a PR against the platform repo.
+
+```shell
+
+# assuming you have already a fork of the platform repo
+# (remote origin pointing to the fork).
+
+# create a new branch to onboard new tenant
+$ git checkout -b onboarding-tenant-a
+
+$ cd tenants
+$ kpt pkg get <pkg-catalog-repo>/tenant@v0.1 tenant-a
+
+# tenant customizations:
+# change the namespace to tenant-a in the tenant-a/Kptfile
+# configure the quota.yaml if needed or
+# or cp .snippets/quota.{small, large, medium}.yaml as quota.yaml
+# cp .snippets/isolate-network.yaml if isolation is needed
+# or directly edit the resources
+
+$ kpt fn render tenant-a
+
+# if all invariants passed, then we are all set.
+
+$ git commit -am "added tenant-a"
+$ git push origin onboarding-tenant-a
+
+# make a pull request for platform team to merge
+# TODO (link to an example PR will be great here)
+```
+
+## Day 2 use-cases
+
+Platform team can evolve the tenant package over time for example, introducing
+additional invariants to be enforced, updating the defaults for quota etc. So
+assuming a new version of the tenant package `tenant/v0.2` has been published.
+Let’s go through the steps needed to update the tenant package.
+
+```shell
+
+# assuming you have already a fork of the platform repo
+# (remote origin pointing to the fork).
+
+# create a new branch to update tenant
+$ git checkout -b update-tenant-a
+
+$ kpt pkg update tenant-a@tenants/v0.2
+
+$ kpt fn render tenant-a
+
+# if all invariants passed, then we are all set.
+
+$ git commit -am "updated tenant-a to newer version"
+$ git push origin update-tenant-a
+
+# make a pull request for platform team to merge
+# TODO (link to an example PR will be great here)
+
+// TODO: explain the edge cases where `pkg update` may not be smooth.
+
+```
+
+Note that the platform team can add/update the snippets in the tenant package.
+`kpt pkg update` updates the hidden snippets directory automatically but
+resources created by copying snippets will not be updated automatically and
+need to be updated manually if needed.
```

Comment:
If user copies over the resource, the identity(merge-comment) is also copied. So technically the merge should be automatic even for the copied resource. But merge behavior takes file-path also as identity. It is worth exploring special casing resources in `.snippets` directory so that this problem can be solved completely.

Reply:
Actually, as per current convention, `.snippets` directory is included in `.krmignore` to instruct kpt to ignore it, so I don't think merge-comment will be added to it when we fetch the pkg.

`snippets` is exploration at this point. If the pattern resonates with users, we can definitely support natively but at this point, it will be premature to special case it.